addons/account/models/account_move.py

```diff
@@ -382,14 +382,24 @@ def _get_accounting_date(self, invoice_date, has_tax):
         :param has_tax (bool): Iff any taxes are involved in the lines of the invoice
         :return (datetime.date):
         """
-        tax_lock_date = self.company_id.tax_lock_date
         today = fields.Date.today()
-        if invoice_date and tax_lock_date and has_tax and invoice_date <= tax_lock_date:
-            invoice_date = tax_lock_date + timedelta(days=1)
 
         if self.is_sale_document(include_receipts=True):
+            closest_lock_date, lock_type = self._get_closest_violated_lock_date_and_type(invoice_date, has_tax)
```

Comment:
How come we are checking both locks for sales but only the tax lock for purchases?
We are trying to fix importing invoices coming from another software here, which is "exactly" the same as importing vendor bills